### ISSUE TITLE
ScanNet scene results

### ISSUE THREAD

**Issue by ZheningHuang** [manycore-research/SpatialLM#35]
<img width="795" alt="Image" src="https://github.com/user-attachments/assets/b3eb3805-e171-4181-9fb1-163fdecc0f98" />

Hi, I tested on a ScanNet scene, and the results look a bit odd.

Any suggestions? All I did was put the scannet.ply into the model.

Thanks.

**Comment by SamMaoYS** [comment]
Hi, the model takes axis aligned point clouds as inputs, the ply file in ScanNet is not axis aligned, I think you need to apply the ScanNet alignment matrix to the point cloud to transform them to axis aligned for better results. Check out section 2 [here](https://github.com/manycore-research/SpatialLM/blob/main/EXAMPLE.md#2-align-the-output-point-cloud).

**Comment by ZheningHuang** [comment]
Hi,

It still doesn’t seem ideal:

<img width="903" alt="Image" src="https://github.com/user-attachments/assets/aabe7283-3fa1-46b8-8d82-4baea1651b47" />

Regardless of the alignment issue, the results still do not appear ideal. There are many mismatches and detections that seem to occur out of nowhere.

Maybe I have missed some key steps—please let me know.

Thanks.

**Comment by SamMaoYS** [comment]
Hi, yes this is indeed odd.

I tested with the same scene0000_00.ply, here is my test result.

![Image](https://github.com/user-attachments/assets/17fa5c42-8945-4c7d-98b7-8e14363d70f3)

All I did, is apply the alignment matrix to the original scene0000_00_vh_clean_2.ply, and save to a aligned ply mesh file.
Then run: 
```python
python inference.py --point_cloud scene0000_00.ply --output scene0000_00.txt
```

The results are not perfect and still contain mispredictions. As this scene differs a lot from our training set, which primarily consists of typical Chinese residential apartments, including living rooms, bedrooms, kitchen and bathrooms. Since this scene in particular is a mix of four room types, our model struggles to perform well. This is a weakness of the current model that requires further improvement. However, we expect that at least the architectural wall elements should work relatively well and align correctly with the point cloud positions.

Could you provide more details about your test settings? We can then go through the process together to check if anything is missing or not well documented. This will help us improve the documentation, making it easier for more users to test our models on their own cases.

Thank you!

**Comment by shuttworth** [comment]
How will the original video and 3D boxes be in one video?

**Comment by alan0717-cn** [comment]
Hello, I would like to know how to apply an alignment matrix to point cloud data. I am currently using U-ARE-ME.
